packages/rocketchat-lib/server/methods/getChannelHistory.js

```diff
@@ -0,0 +1,83 @@
+Meteor.methods({
+	getChannelHistory({rid, latest, oldest, inclusive, count = 20, unreads}) {
+		check(rid, String);
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'getChannelHistory' });
+		}
+
+		const fromUserId = Meteor.userId();
+		const room = Meteor.call('canAccessRoom', rid, fromUserId);
+		if (!room) {
+			return false;
+		}
+
+		//Make sure they can access the room
+		if (room.t === 'c' && !RocketChat.authz.hasPermission(fromUserId, 'preview-c-room') && room.usernames.indexOf(room.username) === -1) {
+			return false;
+		}
+
+		//Ensure latest is always defined.
+		if (_.isUndefined(latest)) {
+			latest = new Date();
+		}
+
+		//Verify oldest is a date if it exists
+		if (!_.isUndefined(oldest)) {
+			if (!_.isDate(oldest)) {
```

Comment:
Don't you think this will be better to read as `if (!_.isUndefined(oldest) && !_.isDate(oldest)) {`?